lib/jira/resource/issue.rb

```diff
@@ -40,12 +40,21 @@ class Issue < JIRA::Base
       has_many :remotelink, :class => JIRA::Resource::Remotelink
 
       def self.all(client)
-        url = client.options[:rest_base_path] + "/search?expand=transitions.fields"
-        response = client.get(url)
-        json = parse_json(response.body)
-        json['issues'].map do |issue|
-          client.Issue.build(issue)
+        start_at = 0
+        max_results = 1000
+        result = []
+        loop do
+          url = client.options[:rest_base_path] +
+                "/search?expand=transitions.fields&maxResults=#{max_results}&startAt=#{start_at}"
+          response = client.get(url)
+          json = parse_json(response.body)
+          json['issues'].map do |issue|
+            result.push(client.Issue.build(issue))
+          end
+          break if json['issues'].size == 0
+          start_at += json['issues'].size
```

Comment:
In that loop, instead of making a request each time to see if there's data still available, I can leverage the use of the meta data contained in the response from jira

```
"expand": "schema,names",
"startAt": 0,
"maxResults": 1000,
"total": 11
```

So if `startAt + total` is greater then `maxResults` I make the request if not, no request.

Then after that I can create a new test to test that spec.